### PR TITLE
rocksdb: Upgrade rocksdb

### LINKED ISSUE #17617: Upgrade RocksDB to 8.10
## Development Task
Upgrade RocksDB library to 8.10

### PR DESCRIPTION
**Pull request by v01dstar**
### What is changed and how it works?

Issue Number: Close #17617

What's Changed:

```commit-message
Upgrade RocksDB to 8.10

* Deprecate RocksEngineIterator::sequence()
* Deprecate seq in CompactionFilter, since it is no longer used by TiKV
* Some statistics have been removed from RocksDB, accommodate that change in TiKV
* Dynamic_level_bytes is now default true, some tests are modified to reflect this change
* Removing test_rocksdb_listener, because https://github.com/facebook/rocksdb/pull/9758 overwrited https://github.com/tikv/rocksdb/pull/347, and the test is only for raftstore v2 which is not being used.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Performance regression: Throughput drop
- [ ] Breaking backward compatibility

### Release note

```release-note
Upgrading RocksDB to 8.10. It is expected to see slight performance regression (<5%) in certain use cases.
```

### CODE REVIEW

File: components/engine_rocks/src/rocks_metrics.rs [hunk 1]
```diff
@@ -1588,31 +1524,6 @@ lazy_static! {
...
-    pub static ref STORE_ENGINE_HARD_RATE_LIMIT_DELAY_VEC: GaugeVec = register_gauge_vec!(
-        "tikv_engine_hard_rate_limit_delay_count",
-        "Histogram of hard rate limit delay count",
-        &["db", "type"]
-    ).unwrap();
-    pub static ref STORE_ENGINE_SOFT_RATE_LIMIT_DELAY_VEC: GaugeVec = register_gauge_vec!(
-        "tikv_engine_soft_rate_limit_delay_count",
-        "Histogram of soft rate limit delay count",
-        &["db", "type"]
-    ).unwrap();
```
Comment (overvenus) [inline components/engine_rocks/src/rocks_metrics.rs:1615]: Need clean up in tikv_details as well?

Comment (overvenus) [inline components/engine_rocks/src/rocks_metrics.rs:1615]: Clean up in tikv_details as well?

Comment (v01dstar) [reply components/engine_rocks/src/rocks_metrics.rs:1615]: I don't see them in tikv_details. Can you point me to the place?

File: tests/failpoints/cases/test_engine.rs [hunk 2]
```diff
@@ -56,86 +51,3 @@ fn test_write_buffer_manager() {
...
-//                                            t1(0-10), t2(10-20)
-//                                            flush job(0-20)
-// 6.                                           finish flush
-// 7.                                           unlock
-// 8.                                           on_flush_completed:
-//                                            update last_flushed to 20
-// 9.       on_memtable_sealed
-//        10 > 20 *panic*
-// ```
-#[test]
```
[the remark this answers is not in the document]
Comment (v01dstar) [reply tests/failpoints/cases/test_engine.rs:82]: Done

**Review by Connor1996 — APPROVED** [review]

File: src/server/gc_worker/compaction_filter.rs [hunk 3]
```diff
@@ -263,7 +263,7 @@ impl CompactionFilterFactory for WriteCompactionFilterFactory {
 
         debug!(
             "gc in compaction filter"; "safe_point" => safe_point,
-            "files" => ?context.file_numbers(),
+            "files" => ?context.input_table_properties().iter().map(|(k, _)| k).collect::<Vec<_>>(),
```
[the remark this answers is not in the document]
Comment (LykxSassinator) [reply src/server/gc_worker/compaction_filter.rs:266]: +1

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**